impedancefitter/fitter.py

```diff
@@ -134,10 +135,10 @@ class Fitter(object):
         Exists only when :meth:`run` was called.
     """
 
-    def __init__(self, inputformat, directory=None, **kwargs):
+    def __init__(self, inputformat, directory=None, df=None, **kwargs):
         """ initializes the Fitter object
         """
-
+        self.df = df
```

Comment:
I would not add the dataframe to the attributes of the Fitter object.
Just use it in the entire function.